samples/knn/knn-lucene-filter.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python
+
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import os
+import random
+
+from opensearchpy import OpenSearch, helpers
+
+# connect to an instance of OpenSearch
+
+host = os.getenv('HOST', default='localhost')
+port = int(os.getenv('PORT', 9200))
+auth = (
+    os.getenv('USERNAME', 'admin'), 
+    os.getenv('PASSWORD', 'admin')
+)
+
+client = OpenSearch(
+    hosts = [{'host': host, 'port': port}],
+    http_auth = auth,
+    use_ssl = True,
+    verify_certs = False,
+    ssl_show_warn = False
+)
+
+# check whether an index exists
+index_name = "hotels-index"
+
+if not client.indices.exists(index_name):
+    client.indices.create(index_name, 
+        body={
+            "settings":{
+                "index.knn": True,
+                "knn.algo_param.ef_search": 100,
+                "number_of_shards": 1,
+                "number_of_replicas": 0
+            },
+            "mappings":{
+                "properties": {
+                    "location": {
+                        "type": "knn_vector", 
+                        "dimension": 2,
+                        "method": {
+                            "name": "hnsw",
+                            "space_type": "l2",
+                            "engine": "lucene",
```

Comment:
the same filter with 2.9 launch is supported with Faiss also, lets add that example too. Also, lets rename the file to efficient filters.

Reply:
I renamed the file.